pandas/core/strings.py

```diff
@@ -2325,12 +2206,110 @@ def cat(self, others=None, sep=None, na_rep=None, join=None):
             res = Series(res, index=data.index, name=self._orig.name)
         return res
 
-    @copy(str_split)
+    _shared_docs['str_split'] = ("""
+    Split strings around given separator/delimiter.
+
+    Splits the string in the Series/Index from the %(side)s,
+    at the specified delimiter string. Equivalent to :meth:`str.%(method)s`.
+
+    Parameters
+    ----------
+    pat : str, optional
+        String or regular expression to split on.
+        If not specified, split on whitespace.
+    n : int, default -1 (all)
+        Limit number of splits in output.
+        ``None``, 0 and -1 will be interpreted as return all splits.
+    expand : bool, default False
+        Expand the splitted strings into separate columns.
+
+        * If ``True``, return DataFrame/MultiIndex expanding dimensionality.
+        * If ``False``, return Series/Index, containing lists of strings.
+
+    Returns
+    -------
+    Series, Index, DataFrame or MultiIndex
+        Type matches caller unless ``expand=True`` (see Notes).
+
+    See Also
+    --------
+     Series.str.split : Split strings around given separator/delimiter.
+     Series.str.rsplit : Splits string around given separator/delimiter, 
+     starting from the right.
+     Series.str.join : Join lists contained as elements in the Series/Index 
+     with passed delimiter. 
+     str.split : Standard library version for split.
+     str.rsplit : Standard library version for rsplit.
+
+    Notes
+    -----
+    The handling of the `n` keyword depends on the number of found splits:
+
+    - If found splits > `n`,  make first `n` splits only
+    - If found splits <= `n`, make all splits
+    - If for a certain row the number of found splits < `n`,
+      append `None` for padding up to `n` if ``expand=True``
+
+    If using ``expand=True``, Series and Index callers return DataFrame and
+    MultiIndex objects, respectively.
+
+    Examples
+    --------
+    >>> s = pd.Series(["this is a regular sentence", "path/to/python/file", np.nan])
+
+    By default, split and rsplit will return an object of the same size
+    having lists containing the split elements.
+
+    >>> s.str.split()
+    0    [this, is, a, regular, sentence]
+    1               [path/to/python/file]
+    2                                 NaN
+    dtype: object
+
+    Parameter `n` can be used to limit the number of splits on the
+    delimiter.
+
+    >>> s.str.split("/", n=2) 
+    0    [this is a regular sentence]
+    1         [path, to, python/file]
+    2                             NaN
+    dtype: object 
+
+    >>> s.str.rsplit("/", n=2)
+    0    [this is a regular sentence]
+    1         [path/to, python, file]
+    2                             NaN
+    dtype: object
```

Comment:
Looks good to me. May be it would be a bit more clear if we introduce one parameter at a time, each with its explanation. Meaning that these two previous examples will be these 3:
- `s.str.split(pat='/')` - with a sentence explaining that `pat` can be used to split by other characters
- `s.str.split(n=2)` - that's the one you're explaining, but I'd use the default `pat` to make it slightly simple
- `s.str.rsplit(n=2) - here we can explain how `rsplit` is different than `split`.

That does not fully address your concern that if can feel strange that `rsplit` page starts by a `split` example, but IMO will make it easier to understand when `rsplit` should be used.

An idea that just came to my mind is that instead of `/path/to/python/file` we could use for example `http://pandas.pydata.org/pandas-docs/stable/api.html` (or a shorter url). Then, `rsplit(n=1)` can be used to split the html document from the rest of the url.

One last idea that you can consider is having an example `s.str.rsplit()` after the first one. With a comment like "Without the `n` parameter, `rsplit` behaves like `split`" (I think this is the case, may be you want to double check, for example when using `expand`). If you think this can make users in the `rsplit` page be less confused, it could be a good idea.

The docstring looks good to me, feel free to incorporate the changes that makes sense to you.

Reply:
I like the idea of using http://pandas.pydata.org/pandas-docs/stable/api.html but then the issue is that we will need to rsplit(pat ='/', n=1) as rsplit(n=1) won't work in this instance -should this be another example then as its utilising both parameters pat and n ? So how i see the different examples ordered and sectioned are as follows - 
Ive seen other docstrings only explain the examples at the beginning of sections (where a different parameter is introduced) - so in this case where i will explain how `rsplit` is different to `split` - does this have to be introduced at the beginning when introducing both `split` and `rsplit` examples for `n=2` or can the sentence come just before the `rsplit` example ?

## Default example 
* `s.str.split()`
* `s.str.rsplit()`  - sentence explaining how `rsplit` produces the same output as `split` in this case 
## Parameter n 
* `s.str.split(n=2)`
* `s.str.rsplit(n=2)` - explain how `rsplit` is different than `split`
## Parameter pat 
* `s.str.split(pat=' / ')` - sentence explaining how `pat` can be used to split by other characters like `/`
## Real world example using both pat and n 
* `s.str.rsplit(pat = ' / ', n=1)` - example of how pat and n can be used together to split the html document in `"http://pandas.pydata.org/pandas-docs/stable/visualization.html"` from rest of url.